unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PerformanceMeterController/PerformanceMeterController.cs

```diff
@@ -0,0 +1,284 @@
+                + "\n * Quality Settings -> Shadow Distance -> " + Settings.i.currentQualitySettings.shadowDistance
+                + "\n * Quality Settings -> Shadow Resolution -> " + Settings.i.currentQualitySettings.shadowResolution
+                + "\n * Quality Settings -> Soft Shadows -> " + Settings.i.currentQualitySettings.softShadows
+                + "\n * Quality Settings -> SSAO Quality -> " + Settings.i.currentQualitySettings.ssaoQuality
+                + "\n * Quality Settings -> Camera Draw Distance -> " + Settings.i.currentQualitySettings.cameraDrawDistance
+                + "\n * Quality Settings -> Detail Object Culling Enabled -> " + Settings.i.currentQualitySettings.enableDetailObjectCulling
+                + "\n * Quality Settings -> Detail Object Culling Limit -> " + Settings.i.currentQualitySettings.detailObjectCullingLimit
+            );
+
+            // Step 2 - report processed data
+            Log("Data report step 2 - Processed values:"
+                + "\n * PERFORMANCE SCORE (0-10) -> " + CalculatePerformanceScore()
+                + "\n * average FPS -> " + averageFPS
+                + "\n * highest FPS -> " + highestFPS
+                + "\n * lowest FPS -> " + lowestFPS
+                + "\n * 50 percentile (median) FPS -> " + percentile50FPS
+                + "\n * 95 percentile FPS -> " + percentile95FPS
+                + "\n * total hiccups -> " + totalHiccupFrames
+                + "\n * total hiccups time (seconds) -> " + totalHiccupsTimeInSeconds
+                + "\n * total frames -> " + totalFrames
+                + "\n * total frames time (seconds) -> " + totalFramesTimeInSeconds
+            );
+
+            // Step 3 - report all samples data
+            string rawSamplesJSON = "Data report step 3 - Raw samples:"
+                                    + "\n "
+                                    + "{\"frame-samples\": " + JsonConvert.SerializeObject(samples) + "}";
+
+#if !UNITY_WEBGL
+            string targetFilePath = Application.persistentDataPath + "/PerformanceMeterRawFrames.txt";
+            Log("Data report step 3 - Trying to dump raw samples JSON at: " + targetFilePath);
+            System.IO.File.WriteAllText (targetFilePath, rawSamplesJSON);
+#endif
+
+            Log(rawSamplesJSON);
+        }
+
+        /// <summary>
+        /// Calculates a performance score from 0 to 10 based on the average FPS (compared to the max possible FPS) and the amount of hiccup frames (compared to the total amount of frames).
+        /// </summary>
+        private float CalculatePerformanceScore()
+        {
+            float topFPS = Settings.i.currentQualitySettings.fpsCap ? 30f : 60f;
+            float fpsScore = Mathf.Min(averageFPS / topFPS, 1); // from 0 to 1
+            float hiccupsScore = 1 - totalHiccupFrames / samples.Count; // from 0 to 1
+            float performanceScore = (fpsScore + hiccupsScore) / 2 * 10; // scores sum / amount of scores * 10 to have a 0-10 scale
```

Comment:
I think that restricting the performance score to 0-10 may not give enough granularity. This score should help us to diagnose improvements to small performance tweaks. I think this is the reason PCMark scores go above 1000 to cite an example. I suggest --at the very least-- to up this value to 0-100 with the same number of decimals.

Reply:
great idea, done!